src/main/java/de/dhbw/music/ShapeProcessor.java

```diff
@@ -0,0 +1,119 @@
+package de.dhbw.music;
+
+import de.dhbw.video.shape.Shape;
+import de.dhbw.video.shape.ShapeForm;
+import de.dhbw.video.shape.ShapeType;
+import org.opencv.core.Core;
+import org.opencv.core.CvType;
+import org.opencv.core.Mat;
+
+import java.util.List;
+
+import static de.dhbw.statics.*;
+
+public class ShapeProcessor {
+
+    List<Shape> shapes;
+    // consists of x_pos, y_pos, width, height, is_rect(0/1)
+    int[] playfieldInfo;
+    Mat[] playFieldBoundaries;
+    int frameHeight;
+    int frameWidth;
+    boolean[][] soundMatrix;
+    public ShapeProcessor(){
+        playFieldBoundaries = new Mat[4];
+        for(int i = 0; i < 4; i++) {
+            playFieldBoundaries[i] = new Mat(2,1, CvType.CV_32SC1);
+        }
+        playfieldInfo = new int[5];
+    }
+
+    //TODO check if making the width and height of  the input-video globally available makes sense
+    public void processShapes(List<Shape> shapes, int width, int height){
+        this.shapes = shapes;
+        frameWidth = width;
+        frameHeight = height;
+        detectPlayfield();
+        if(playfieldInfo[4] == 1){
+            generateSoundMatrix();
+        }
+        // TODO > treat control markers
+    }
+
+    public boolean[][] getSoundMatrix(){
+        return soundMatrix;
+    }
+
+    private void detectPlayfield(){
+        playfieldInfo[4] = 1;
+        Shape[] squares = shapes.stream().filter(shape -> shape.getForm() == ShapeForm.SQUARE).toArray(Shape[]::new);
+        if(squares.length < 4){
+            playfieldInfo[4] = 0;
+            System.err.println("Not enough squares for playfield");
+        }
+        else {
+            Shape[] corners = new Shape[]{squares[0], squares[1], squares[2], squares[3]};
+            for (Shape square : squares) {
+                // top left
+                if(Math.sqrt(square.pos[1] * square.pos[1] + square.pos[0] * square.pos[0]) < Math.sqrt(corners[0].pos[0] * corners[0].pos[0] + corners[0].pos[1] * corners[0].pos[1])){
```

Comment:
Sehr cool, dass hier casual Pythagoras auftaucht!Kann ich dich überzeugen die Berechnungen in ne Variable auszulagern für Lesbarkeit? 😇  Generell für die vier Ecken, bei den anderen wird es ja auch ein ganz bisschen komplizierter

Reply:
so halb, eine Variable fände ich ein bisschen wenig und bei mehreren kommt mir das vor wie unnötiger Rechen/Variablen-Zuweisungs-Overhead

Reply:
Ich hab einfach mal Kommentare spendiert :sweat_smile: